SpatialGDK/Source/Private/Interop/SpatialSender.cpp

```diff
@@ -0,0 +1,585 @@
+			ChannelToUnresolved.Remove(ChannelObjectPair);
+			if (ChannelToUnresolved.Num() == 0)
+			{
+				ObjectToUnresolved.Remove(UnresolvedObject);
+			}
+		}
+	}
+
+	HandleToUnresolved->Remove(Handle);
+	if (HandleToUnresolved->Num() == 0)
+	{
+		PropertyToUnresolved.Remove(ChannelObjectPair);
+	}
+}
+
+void USpatialSender::QueueOutgoingUpdate(USpatialActorChannel* DependentChannel, UObject* ReplicatedObject, int16 Handle, const TSet<const UObject*>& UnresolvedObjects, bool bIsHandover)
+{
+	check(DependentChannel);
+	check(ReplicatedObject);
+	FChannelObjectPair ChannelObjectPair(DependentChannel, ReplicatedObject);
+
+	UE_LOG(LogTemp, Log, TEXT("Added pending outgoing property: channel: %s, object: %s, handle: %d. Depending on objects:"),
+		*DependentChannel->GetName(), *ReplicatedObject->GetName(), Handle);
+
+	// Choose the correct container based on whether it's handover or not
+	FChannelToHandleToUnresolved& PropertyToUnresolved = bIsHandover ? HandoverPropertyToUnresolved : RepPropertyToUnresolved;
+	FOutgoingRepUpdates& ObjectToUnresolved = bIsHandover ? HandoverObjectToUnresolved : RepObjectToUnresolved;
+
+	FUnresolvedEntry Unresolved = MakeShared<TSet<const UObject*>>();
+	*Unresolved = UnresolvedObjects;
+
+	FHandleToUnresolved& HandleToUnresolved = PropertyToUnresolved.FindOrAdd(ChannelObjectPair);
+	check(!HandleToUnresolved.Find(Handle));
+	HandleToUnresolved.Add(Handle, Unresolved);
+
+	for (const UObject* UnresolvedObject : UnresolvedObjects)
+	{
+		FHandleToUnresolved& AnotherHandleToUnresolved = ObjectToUnresolved.FindOrAdd(UnresolvedObject).FindOrAdd(ChannelObjectPair);
+		check(!AnotherHandleToUnresolved.Find(Handle));
+		AnotherHandleToUnresolved.Add(Handle, Unresolved);
+
+		// Following up on the previous log: listing the unresolved objects
+		UE_LOG(LogTemp, Log, TEXT("%s"), *UnresolvedObject->GetName());
+	}
+}
+
+void USpatialSender::QueueOutgoingRPC(const UObject* UnresolvedObject, UObject* TargetObject, UFunction* Function, void* Parameters)
+{
+	check(UnresolvedObject);
+	UE_LOG(LogTemp, Log, TEXT("Added pending outgoing RPC depending on object: %s, target: %s, function: %s"), *UnresolvedObject->GetName(), *TargetObject->GetName(), *Function->GetName());
+	OutgoingRPCs.FindOrAdd(UnresolvedObject).Add(FPendingRPCParams(TargetObject, Function, Parameters));
+}
+
+Worker_CommandRequest USpatialSender::CreateRPCCommandRequest(UObject* TargetObject, UFunction* Function, void* Parameters, Worker_ComponentId ComponentId, Schema_FieldId CommandIndex, Worker_EntityId& OutEntityId, const UObject*& OutUnresolvedObject)
+{
+	Worker_CommandRequest CommandRequest = {};
+	CommandRequest.component_id = ComponentId;
+	CommandRequest.schema_type = Schema_CreateCommandRequest(ComponentId, CommandIndex);
+	Schema_Object* RequestObject = Schema_GetCommandRequestObject(CommandRequest.schema_type);
+
+	improbable::UnrealObjectRef TargetObjectRef(PackageMap->GetUnrealObjectRefFromNetGUID(PackageMap->GetNetGUIDFromObject(TargetObject)));
```

Comment:
Most of this function is duplicated `CreateMulticastUpdate`, extract similar functionality?